### ISSUE TITLE
MariaDB not working properly with Django 3.0

### ISSUE THREAD

**Issue by lpulley** [uiuc-mrdc/cogs#13]
In the `django3` branch, I've configured `docker-compose.yml` to use MariaDB instead of MongoDB, since MariaDB is now officially supported through the `mysql` stuff in Django, and Djongo appears to not work on Django 3.0 yet. Would be nice to move to first-party-supported databases anyway.

However, MariaDB takes some time to initialize itself on first start, and the project needs to be restarted before Django properly connects to it. I don't think the solution for this is bash scripts; there *should* be a clean first-party Docker/MariaDB/Django solution to this. I can look further into this over winter break.

**Comment by lpulley** [comment]
Actually, in its current state, it seems that Django can't connect at all to MariaDB. This definitely needs work and it's my top priority when I get time for this.

**Comment by lpulley** [comment]
Actually, in its current state, it seems that Django can't connect at all to MariaDB. This definitely needs work and it's my top priority when I get time for this.

**Comment by lpulley** [comment]
Actually, in its current state, it seems that Django can't connect at all to MariaDB. This definitely needs work and it's my top priority when I get time for this.

**Comment by lpulley** [comment]
![image](https://user-images.githubusercontent.com/7193187/70830971-077bdc80-1db7-11ea-9854-8b7240bb0462.png)

MariaDB's Docker documentation seems to be aware of this issue.

**Comment by mgale456** [comment]
[This stackexchange answer by Thalik](https://stackoverflow.com/questions/54479941/how-to-init-mysql-database-in-docker-compose) got me hopeful for a simple first party thing in `docker-compose.yml`. Apparently what he does in his `docker-compose.yml` is no longer supported. 

This lead me to read the [docker-compose file documentation](https://docs.docker.com/compose/compose-file/). The section for `depends_on` lead me to [this page](https://docs.docker.com/compose/startup-order/), which is about controlling startup order.

TL;DR Docker says it is best to have a resilient app that checks and reconnects itself if the db doesn't connect, but something like `wait-for-it.sh` is a decent alternative if you don't need that resilience

Also, I haven't searched much at all for ways to do this in Django, but I didn't find it at a glance.

**Comment by lpulley** [comment]
Yeah, it does sound like having Django do this would be the cleanest way. I'll look into have it retry the database connection.

**Comment by lpulley** [comment]
This one might not retry enough but seems like a clean solution if it does work: https://github.com/udemy/django-dbconn-retry 

Can install with `requirements.txt` and `INSTALLED_APPS`.

**Comment by mgale456** [comment]
I tried that app, and it did not fix the startup. I also reverted that commit, so you can go back and see them if  you want.

In browsing for another potential solution for this, I saw that it is the runserver command that is halting. We may want to see how Daphne handles startup, and it if hang up on the same db checks

**Comment by lpulley** [comment]
I've found that PostgreSQL takes a few seconds to initialize instead of a minute. Seems like PostgreSQL has the "cool kid" factor too, might be something we want to use for this project, especially if it plays more nicely with Docker.

**Comment by mgale456** [comment]
PostgreSQL seems to be supported by core Django unlike MongoDB was, so it seems like a fine candidate to me. idk anything else about it, and I'm not sure if there could really be any problems caused by it, since it is also a full featured database afaik. 

If it improves our setup instructions by 2 steps, I'm all in. That's very nice for the future ease of CoGS.
